pkg/util/conditions/condition_test.go

```diff
@@ -0,0 +1,226 @@
+package conditions
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/clock"
+
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
+	aroclient "github.com/Azure/ARO-RP/pkg/operator/clientset/versioned"
+	arofake "github.com/Azure/ARO-RP/pkg/operator/clientset/versioned/fake"
+	"github.com/Azure/ARO-RP/pkg/util/cmp"
+)
+
+func TestSetCondition(t *testing.T) {
+	ctx := context.Background()
+	role := "master"
+	objectName := "cluster"
+
+	kubeclock = &clock.FakeClock{}
+	var transitionTime metav1.Time = metav1.Time{Time: kubeclock.Now()}
+
+	for _, tt := range []struct {
+		name      string
+		aroclient aroclient.Interface
+		input     corev1.PodCondition
+
+		expected []corev1.PodCondition
+		wantErr  error
+	}{
+		{
+			name: "noop",
```

Comment:
This might benefit fro better naming. Why noop?

Reply:
This actually triggers an update because of:

```go
	if role == operator.RoleMaster && cluster.Status.OperatorVersion != version.GitCommit {
		cluster.Status.OperatorVersion = version.GitCommit
		changed = true
	}
```

Which highlights that `SetCondition` name is confusing & that it probably doing more than it should.

Reply:
yap. It treats version as condition. 
Not much we can do here (not the place too). 
Added test to over this for now